src/System.Buffers/src/System/Buffers/NativeBufferBucket.cs

```diff
@@ -33,13 +34,26 @@ internal NativeBufferBucket(int elementsInBuffer, int numberOfBuffers)
             }
         }
 
-        ~NativeBufferBucket()
```

Comment:
Got it...updated to add the finalizers + call to Dispose(false) and made the NativeBufferBucket always call FreeHGlobal